### PR TITLE
LIME-354: + add advertisement banners to pages

### LINKED ISSUE #354: Add Advertisements
- [ ] Display Advertisements for unsubscribed users
- [ ] Do not show Advertisements to subscribed users

ADS PLACEMENT
1. For small screens display ads between the sections of the app, user will see them while scrolling
Image

2. Ads can be displayed between the sections of the content on the big screens like so:
Image

3. For bigger screens add columns on the sides from the original design:
Image

### PR DESCRIPTION
**Pull request by ivanbilobrov07**
image

image

image

### CODE REVIEW

**Comment by nikita-remeslov** [comment]
@ivanbilobrov07 Could you add mobile view screens?

**Comment by ivanbilobrov07** [comment]
At the moment, there is 1 in the home page and 1 in the goals

image

image

**Comment by ivanbilobrov07** [comment]
Also, I just noticed that they have different height, so I will fix it now

File: frontend/src/bundles/goals/pages/goals.tsx [hunk 1]
```diff
@@ -104,107 +109,115 @@ const Goals: React.FC = () => {
...
+                        <div className="flex w-full flex-col gap-8 xl:w-[70%]">
+                            <section>
+                                <GoalWidget
+                                    value={
+                                        (lastGoal?.distance as number) ||
+                                        (lastGoal?.duration as number)
+                                    }
+                                    target={
+                                        (lastGoal?.distance as number) ||
+                                        (lastGoal?.duration as number)
```
Comment (jhladkov) [inline frontend/src/bundles/goals/pages/goals.tsx]: You can write it once and store in variable. 
`const YOUR_VARIABLE_NAME= (lastGoal?.distance as number) || (lastGoal?.duration as number);`

```
value={YOUR_VARIABLE_NAME}
target={YOUR_VARIABLE_NAME}
```

**Review by dimapopovych — DISMISSED** [review]

File: frontend/src/bundles/common/components/google-ads/google-ads.tsx [hunk 2]
```diff
@@ -1,10 +1,18 @@
+import { getValidClassNames } from '~/bundles/common/helpers/helpers.js';
+
 type GoogleAdsProperties = {
```
Comment (stpalamar) [inline frontend/src/bundles/common/components/google-ads/google-ads.tsx]:
```suggestion
type Properties = {
```

**Review by stpalamar — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/base-layout/base-layout.tsx [hunk 3]
```diff
@@ -3,8 +3,8 @@ import { Sidebar } from '~/bundles/common/components/sidebar/sidebar.js';
 import { type ReactNode } from '~/bundles/common/types/types.js';
 
 import { getValidClassNames } from '../../helpers/helpers.js';
```
Comment (SofiiaTrokhymchuk) [inline frontend/src/bundles/common/components/base-layout/base-layout.tsx]: Please, use proper imports

File: frontend/src/bundles/common/components/base-layout/base-layout.tsx [hunk 4]
```diff
@@ -3,8 +3,8 @@ import { Sidebar } from '~/bundles/common/components/sidebar/sidebar.js';
...
-import { useSidebarToggle } from '../../hooks/hooks.js';
-import { RouterOutlet } from '../components.js';
+import { useAppSelector, useSidebarToggle } from '../../hooks/hooks.js';
```
Comment (SofiiaTrokhymchuk) [inline frontend/src/bundles/common/components/base-layout/base-layout.tsx]: imports

File: frontend/src/bundles/common/components/modal/modal.tsx [hunk 5]
```diff
@@ -12,10 +13,17 @@ type Properties = {
...
+            }
+        },
+        [onClose],
+    );
+
     const classes = {
-        modalClass: `relative transition ease-in-out duration-300 modal z-[2] flex items-center justify-center ${isOpen ? 'visible opacity-100' : 'invisible opacity-0'}`,
-        overlayClass:
-            'overlay bg-primary fixed inset-0 z-50 cursor-pointer opacity-90',
+        overlayClass: `overlay bg-overlay fixed inset-0 z-50 ${isOpen ? 'opacity-1' : 'opacity-0 pointer-events-none'}`,
```
Comment (SofiiaTrokhymchuk) [inline frontend/src/bundles/common/components/modal/modal.tsx]: use getValidClassNames

**Review by SofiiaTrokhymchuk — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**